streaming/aggregate/sum.go

```diff
@@ -41,49 +58,60 @@ func (agg *Sum) AddValue(ctx context.Context, tx storage.StateTransaction, value
 	return nil
 }
 
+func isNeutralElement(currentSum octosql.Value) bool {
+	return currentSum.GetType() == octosql.TypeInt && currentSum.AsInt() == 0
+}
+
+func getAppropriateNeutralElement(valueType octosql.Type) (octosql.Value, error) {
+	switch valueType {
+	case octosql.TypeInt:
+		return octosql.MakeInt(0), nil
+	case octosql.TypeFloat:
+		return octosql.MakeFloat(0.0), nil
+	case octosql.TypeDuration:
+		return octosql.MakeDuration(0), nil
+	default:
+		return octosql.ZeroValue(), errors.Errorf("unsupported value type passed to sum: %s", valueType)
+	}
+}
+
 func (agg *Sum) RetractValue(ctx context.Context, tx storage.StateTransaction, value octosql.Value) error {
 	currentSumPrefix := storage.NewValueState(tx.WithPrefix(currentSumPrefix))
 
+	valueType := value.GetType()
+
 	currentSum, err := agg.GetValue(ctx, tx)
 	if err != nil {
 		return errors.Wrap(err, "couldn't get current sum")
 	}
 
-	clearSum := true
+	// TODO - is this needed ?
```

Comment:
Yes, it's needed because if the retraction comes before the record there may be type error I think